### PR TITLE
Support for Azure OpenAI + Ollama + unit tests for the embeders and LLMs

### PR DESCRIPTION
**Pull request by stellasia**
# Description

- Add `AzureOpenAI` wrappers (LLM and embeddings)
- Add LLMs to `neo4j_graphrag.llm` and embedder to `neo4j_graphrag.embedding` for consistency
- Add tests for OpenAI (LLM + embedder) and VertexAI (embedder)
- Add google cloud aiplatform to dev dependencies (for unit tests)
- Update docs

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [ ] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] Unit tests have been updated
- [ ] E2E tests have been updated
- [x] Examples have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

**Comment by stellasia** [comment]
@CodiumAI-Agent /update_changelog

File: docs/source/user_guide_rag.rst [hunk 1]
```diff
@@ -67,12 +67,44 @@ Each component can be configured individually: the LLM and the prompt.
 Using Another LLM Model
 ========================
 
-If OpenAI can not be used, there are two available alternatives:
+If OpenAI can not be used directly, there are a few available alternatives:
```
Comment (willtai) [inline docs/source/user_guide_rag.rst:70]:
```suggestion
If OpenAI cannot be used directly, there are a few available alternatives:
```

File: src/neo4j_graphrag/embeddings/vertexai.py [hunk 2]
```diff
@@ -38,7 +35,14 @@ class VertexAIEmbeddings(Embedder):
     """
 
     def __init__(self, model: str = "text-embedding-004") -> None:
-        self.vertexai_model = TextEmbeddingModel.from_pretrained(model)
+        if vertexai is None:
```
Comment (alexthomas93) [inline src/neo4j_graphrag/embeddings/vertexai.py:38]: Have fixed this in my PR if you're still having issues

Comment (stellasia) [reply src/neo4j_graphrag/embeddings/vertexai.py:38]: Ok, will rebase after your PR is merged, thanks

**Review by willtai — APPROVED** [review]

**Pull request merged**